.github/workflows/e2e_ppo_trainer.yml

```
@@ -54,6 +35,21 @@ permissions:
   contents: read
 
 jobs:
+  pre_commit_for_ppo:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ["3.12"]
+    steps:
+    - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@0b93645e9fea7318ecaed2b359559ac225c90a2b # v5.3.0
+      with:
+        python-version: ${{ matrix.python-version }}
+    - uses: pre-commit/action@v3.0.1
+      env:
+        RUFF_OUTPUT_FORMAT: github
```

@eric-haibin-lin The following config should work.

```suggestion
      with:
        extra_args: "" # Overriding "--all-files"
        # TODO: Remove this once all the files are fixed, i.e., `pre-commit run --all-files` passes
```